src/index.ts

```diff
@@ -67,41 +75,52 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 		return 1;
 	}
 
-	versionA = normalize(versionA);
-	versionB = normalize(versionB);
+	const nVersionA = normalize(versionA);
+	const nVersionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
+	const isAValid = semver.valid(nVersionA);
+	const isBValid = semver.valid(nVersionB);
 	if (isAValid && !isBValid) {
 		return 1;
 	}
 	if (!isAValid && isBValid) {
 		return -1;
 	}
 	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+		if (nVersionA > nVersionB) {
 			return 1;
 		}
-		if (versionA < versionB) {
+		if (nVersionA < nVersionB) {
 			return -1;
 		}
 		return 0;
 	}
-	const semverResult = semver.compare(versionA, versionB);
+	const semverResult = semver.compare(nVersionA, nVersionB);
 	if (semverResult !== 0) {
 		return semverResult;
 	}
-	const revA = getRev(versionA);
-	const revB = getRev(versionB);
-	if (revA !== revB) {
-		return revA > revB ? 1 : -1;
-	}
-	const devA = Number(isDevelopmentVersion(versionA));
-	const devB = Number(isDevelopmentVersion(versionB));
-	if (devA !== devB) {
-		return devA > devB ? -1 : 1;
+
+	const suffixComparisonMethods = [
+		{ fn: getRev, reverse: false, normalize: true },
+		{ fn: isPreReleaseVariant, reverse: true, normalize: true },
+		{ fn: getPreReleaseVariant, reverse: false, normalize: true },
+		{ fn: getPostReleaseVariant, reverse: false, normalize: false },
```

Comment:
In fact, instead of using unnormalized versions in comparison here, is there a reason we can't just fix this at normalization? We could normalize `1.2.3 (dev)` to `1.2.3+dev`, and simplify later comparisons completely (and get rid of the `versionA`/`nVersionA` stuff en route)

Reply:
I really like this.
Are we fine with changing [this test](https://github.com/resin-io-modules/resin-semver/blob/8727143fa7ed974d3eb932c21fcb73814e161a87/test/index.spec.ts#L446-L454) to have `prod` detected as a `build`?

Reply:
Yes, I think that's fine. That's how we'd normally parse `prod` for an officially correct version number anyway, e.g.

```js
semver.parse('2.3.0+rev1.prod');
```

returns

```json
{
  "raw": "2.3.0+rev1.prod",
  "major": 2,
  "minor": 3,
  "patch": 0,
  "prerelease": [],
  "build": [
    "rev1",
    "prod"
  ],
  "version": "2.3.0"
}
```

Reply:
@pimterry Pushed a new commit based on your feedback, which simplified the code a bit by moving the aforementioned parts in the normalization phase.